python/flask.py

```diff
@@ -31,22 +33,44 @@ def _on_request_tearing_down(sender, **extra):
     del g.mdk_session
 
 
-def mdk_setup(app, timeout=None):
-    """Setup MDK integration with Flask.
+def mdk_setup(app, timeout=None, mdk=None):
+    """
+    Setup MDK integration with Flask.
 
     :param app: A Flask application instance.
     :param timeout: Default timeout in seconds to set for the MDK session.
+    :param mdk: An optional ``mdk.MDK`` instance to use instead of creating a
+        new one. It will not be started or stopped.
 
     :return: The ``mdk.MDK`` instance.
     """
-    app.mdk = mdk.start()
+    if mdk is None:
+        app.mdk = start()
```

Comment:
I gotta admit, I think `mdk.start()` was clearer here. Maybe

```from mdk import start as mdk_start```

? (which is also kind of hideous, but, well...)